pkg/controllers/user/alert/statesyncer/statesyncer.go

```diff
@@ -0,0 +1,138 @@
+package statesyncer
+
+import (
+	"time"
+
+	"github.com/rancher/norman/controller"
+	"github.com/rancher/rancher/pkg/controllers/user/alert/manager"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func NewStateSyncer(cluster *config.UserContext, manager *manager.Manager) *StateSyncer {
+	return &StateSyncer{
+		clusterAlertClient: cluster.Management.Management.ClusterAlerts(cluster.ClusterName),
+		projectAlertClient: cluster.Management.Management.ProjectAlerts(""),
+		alertManager:       manager,
+		clusterName:        cluster.ClusterName,
+	}
+}
+
+type StateSyncer struct {
+	clusterAlertClient v3.ClusterAlertInterface
+	projectAlertClient v3.ProjectAlertInterface
+	alertManager       *manager.Manager
+	clusterName        string
+}
+
+func (s *StateSyncer) Run(stopc <-chan struct{}) {
+
+	tickChan := time.NewTicker(time.Second * 10).C
```

Comment:
You can use util method for ticker with the context: https://github.com/rancher/rancher/blob/master/pkg/controllers/user/healthsyncer/healthsyncer.go#L44

Reply:
ok, I will make changes to all the watcher.